packages/app/src/cli/models/extensions/specifications/flow_template.ts

```diff
@@ -0,0 +1,85 @@
+// import {createUIExtensionSpecification} from '../ui.js'
+import {BaseSchema} from '../schemas.js'
+import {createExtensionSpecification} from '../specification.js'
+import {loadLocalesConfig} from '../../../utilities/extensions/locales-configuration.js'
+import {joinPath} from '@shopify/cli-kit/node/path'
+import {zod} from '@shopify/cli-kit/node/schema'
+import {AbortError} from '@shopify/cli-kit/node/error'
+import {glob} from '@shopify/cli-kit/node/fs'
+import fs from 'fs'
+
+const FlowTemplateExtensionSchema = BaseSchema.extend({
+  name: zod.string(),
+  description: zod.string(),
+  extensions: zod.array(
+    zod.object({
+      handle: zod.string(),
+      type: zod.literal('flow_template'),
+      title: zod.string(),
+      description: zod.string(),
+      categories: zod.array(zod.string()),
+      pre_install_note: zod.string(),
+      post_install_note: zod.string(),
+      require_app: zod.boolean(),
+      visible: zod.boolean(),
+      enabled: zod.boolean(),
+    }),
+  ),
+})
+
+const spec = createExtensionSpecification({
+  identifier: 'flow_template',
+  schema: FlowTemplateExtensionSchema,
+  singleEntryPath: false,
+  appModuleFeatures: (_) => [],
+  deployConfig: async (config, directory) => {
+    // supporting one extension for now in implementation
+    const extension = config.extensions[0]
+    if (!extension) {
+      throw new AbortError(`Missing extension in ${directory} `, 'Make sure you have defined at least one extension.')
+    }
+    if (config.extensions.length > 1) {
+      throw new AbortError(
+        `More than one extension found in ${directory} `,
+        'Make sure you have only one extension defined. This will be expanded to support more in the future.',
+      )
+    }
+    return {
+      handle: extension.handle,
+      type: extension.type,
+      title: extension.title,
+      description: extension.description,
+      categories: extension.categories,
+      pre_install_note: extension.pre_install_note,
+      post_install_note: extension.post_install_note,
+      require_app: extension.require_app,
+      visible: extension.visible,
+      enabled: extension.enabled,
+      localization: await loadLocalesConfig(directory, 'flow_template'),
```

Comment:
How are we handling localization? Where are these saved? Will this be affected by versioning?

Reply:
Captured in the [tech design here](https://docs.google.com/document/d/1APLY_7jPyLV43jmcEw5cVwZqI-OGgouuZ5f2QsNIaiQ/edit#heading=h.zcj4v9c5dxie)

TL;DR:
- Synced to Flow with the payload, stored in a blob in the DB
- Latest published version will override the existing one since there is only ever one version of a Template active at once
- Future work will move these to GCS as the needs of translations grow